packages/template-retail-react-app/app/components/product-tile/index.test.js

```diff
@@ -51,42 +51,71 @@ test('Remove from wishlist cannot be muti-clicked', () => {
     expect(onClick).toHaveBeenCalledTimes(1)
 })
 
-test('Renders variation selection swatches', () => {
+test('Renders variant details based on the selected swatch', async () => {
     const {getAllByRole, getByTestId} = renderWithProviders(
         <ProductTile product={mockProductSearchItem} />
     )
     const swatches = getAllByRole('radio')
     const productImage = getByTestId('product-tile-image')
+    let productTile = getByTestId('product-tile')
 
     // Initial render will show swatched and the image will be the represented product variation
     expect(swatches).toHaveLength(2)
     expect(productImage.firstChild.getAttribute('src')).toBe(
         'https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZRF_001/on/demandware.static/-/Sites-apparel-m-catalog/default/dw175c1a89/images/large/PG.33698RUBN4Q.CHARCWL.PZ.jpg'
     )
+    let currentPriceTag = productTile.querySelectorAll('b')
+    let strikethroughPriceTag = productTile.querySelectorAll('s')
+    expect(currentPriceTag).toHaveLength(1)
+    expect(within(currentPriceTag[0]).getByText(/£191\.99/i)).toBeDefined()
+    expect(strikethroughPriceTag).toHaveLength(1)
+    expect(within(strikethroughPriceTag[0]).getByText(/£320\.00/i)).toBeDefined()
 
-    // Hovering over color swatch changes the image.
-    fireEvent.mouseEnter(swatches[1])
-    waitFor(() => {
-        expect(productImage.firstChild.getAttribute('src')).toBe(
-            'https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZRF_001/on/demandware.static/-/Sites-apparel-m-catalog/default/dw29b7f226/images/large/PG.52002RUBN4Q.NAVYWL.PZ.jpg'
-        )
-    })
+    // Navigating to different color swatch changes the image & price.
+    const swatchGroup = screen.getByRole('radiogroup').parentNode
+    fireEvent.keyDown(swatchGroup, {key: 'ArrowLeft', code: 'ArrowRight', charCode: 37})
```

Comment:
I assume this is the mobile behavior? Since we have different behaviors in mobile and desktop for swatch, it is better to tests both where hovering swatch on desktop, and clicking swatch on mobile

Reply:
This is the desktop behavior. The earlier test that is using `mouseEnter` never worked. For some reason the event is not firing. I followed this [doc](https://testing-library.com/docs/guide-events/) and implemented the alternative - keydown.  Swatch group tests also use keydown event.

Reply:
You meant `mouseOver`? 🤔

Reply:
I tried `mouseOver`as well but didn't work.  mouseEnter was used before my changes

Reply:
My guess why it does not work is because we need to wait for the image to finish loading. Try something like this
```
fire.mouseOver()
// waiting for product-tile-image here won't work because it is already rendered and always be defined. 
// but waiting for the image src here may work since the img src is changing
await waitFor(() => {
  expect(productImage.firstChild.getAttribute('src')).toBe(
        'https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZRF_001/on/demandware.static/-/Sites-apparel-m-catalog/default/dw29b7f226/images/large/PG.52002RUBN4Q.NAVYWL.PZ.jpg'
    )
})
```

Reply:
I believe this is how it was. I will give it a try one more time anyway
Question - Does it matter whether it is a mouseOver or keyDown? All we need is to get the focus (whether is through mouse or key navigation) on the swatch to verify the rendered changes